src/main/java/com/splunk/hecclient/LoadBalancer.java

```diff
@@ -15,26 +15,76 @@
  */
 package com.splunk.hecclient;
 
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.*;
 
 public final class LoadBalancer implements LoadBalancerInf {
+    private static final Logger log = LoggerFactory.getLogger(LoadBalancer.class);
+
     private List<HecChannel> channels;
     private int index;
+    private ConcurrentHashMap<String, List<HecChannel>> indexerInfo; // Map of indexer URI to Channels for that indexer
+    private Set<String> discardedIndexers;
+    // Used for health check
+    private CloseableHttpClient httpClient;
+    private HttpContext context;
+    private HecConfig hecConfig;
+    private boolean keepAlive;
+    private Header[] headers;
+    private String healthCheckEndpoint = "/services/collector/health";
```

Comment:
Not sure if we can rely on HEC responses, such as 503 service unavailable, rather than polling indexers status? To perform this request, we need indexers' Splunk credentials, which is not always available for connectors?

Reply:
One more question: have we tested this endpoint? Say if this endpoint returns OK, does that mean we can ingest data for sure? Is that still possible it has some delay and data ingestion is still failed, and cause data loss?

Reply:
yeah, this is REST API available - https://docs.splunk.com/Documentation/Splunk/8.0.5/RESTREF/RESTinput#services.2Fcollector.2Fhealth

Reply:
we have hec credentials available, they are specified by the user in the config